src/index.js

```diff
@@ -47,15 +49,24 @@ function init (companyId, apiKey, token) {
 class AnalyticsCoreAPI {
   /** Initialize sdk.
   *
-  * @param companyId {string} company ID to be used with Adobe Analytics.
+  * @param companyId {string} company ID to be used with Adobe Analytics (optional - if passed as null, then getDiscoveryCredentials is called to retrieve the company ID).
   * @param apiKey {string} Your api key
   * @param token {string} Valid auth token
+  * @param options {object} Extensible to include additional parameters, as necessary.  For example, can be used to include valid IMS Org ID (i.e., 'F9452D1224FEA71029B5C8A97@AdobeOrg' ) which is only passed when a companyId is unavaiable.
   * @returns {AnalyticsCoreAPI}
   */
-  async init (companyId, apiKey, token) {
+  async init (companyId, apiKey, token, options) {
+    // console.debug(companyId, apiKey, token, imsOrgId)
     const initErrors = []
-    if (!companyId) {
-      initErrors.push('companyId')
+    if (!companyId || companyId === null) {
+      const discoveryResponse = await this.getDiscoveryCredentials(apiKey, token)
+      for (const i in discoveryResponse.imsOrgs) {
+        if (discoveryResponse.imsOrgs[i].imsOrgId === options?.imsOrg) {
```

Comment:
`options?.imsOrg` fails the CI. Optional chaining is only on node 14. We need to support both 10 and 12 still.

Reply:
I've also resolved this in the latest push